packages/docregistry/src/context.ts

```diff
@@ -637,6 +637,43 @@ export class Context<
     }
   }
 
+  /**
+   * Load the metadata of the document without the content.
+   */
+  private _loadContext(): Promise<void> {
+    const opts: Contents.IFetchOptions = {
+      type: this._factory.contentType,
+      content: false,
+      ...(this._factory.fileFormat !== null
+        ? { format: this._factory.fileFormat }
+        : {})
+    };
+    const path = this._path;
+    return this._manager.ready
+      .then(() => {
+        return this._manager.contents.get(path, opts);
+      })
+      .then(contents => {
```

Comment:
I'm seeing another issue here. The image doesn't render into the screen when opening a picture because the widget needs the attribute `format` from the context.
I believe the problem is coming from [here](https://github.com/jupyterlab/jupyterlab/blob/13ca0e0caea5ad943f08627f0c2ed447bc79d829/packages/docregistry/src/context.ts#L654) when requesting the metadata of the file. The server doesn't return the file format.
I've been investigating it, but I can not figure out why.

Reply:
But we don't support collaborative images :smile: 
I mean, `_loadContext` is called when in collaborative mode because we need the metadata of the file without its content, but for an image we actually want the content, right?
Is there some condition to insert [here](https://github.com/jupyterlab/jupyterlab/blob/13ca0e0caea5ad943f08627f0c2ed447bc79d829/packages/docregistry/src/context.ts#L265) for types that are not supported in collaborative mode?

Reply:
I pushed https://github.com/jupyterlab/jupyterlab/pull/12360/commits/d579a7feafe7fc793208a4c383cec6d4ae9c2093, let me know if that's the correct way to do it @hbcarlos.

Reply:
We are sharing every file at the moment. Actually, the implementation that we have in master, is also sharing every file.
For example, an image (`.jpg`) has the type `file` and uses the `DocumentModel` as a model for the data.

The problem is that we have only three types `file`, `notebook` or `directory`. Using the `this._factory.contentType` we can not differentiate between them since a `.txt` and a `.jpg` have the same type. We can not use `this._factory.fileFormat` ("json" | "text" | "base64")  either because I think an image can be loaded as `text` and rendered using its mimetype, see https://github.com/jupyterlab/jupyterlab/blob/a41b9816fdedb936be1589dbc98363f62f5f4574/packages/imageviewer/src/widget.ts#L193-L197

The only solution I can see is by extension, but we will need to explicitly say what extensions should be shared, and I don't think it is a good idea.

Reply:
I pushed https://github.com/jupyterlab/jupyterlab/pull/12360/commits/5708d02d2388b68d24231f351f51e49d41265be5 where we set the room name as `format:type:path`, because the RTC handler needs to load/save with the correct file format.

Reply:
LGTM

Reply:
I'm seeing issues when renaming a notebook a lot of times, it ends up not saving the changes. I need to investigate, maybe it's a Ypy issue with observing changes.

Reply:
Fixed in https://github.com/jupyterlab/jupyterlab/pull/12360/commits/c4c6aa4146eda3d77c22d29bbb3492100471f7c4.

Reply:
There are still small issues with the dirty flag sometimes not being cleared when opening a file. I'm going to add:
```py
self.room.document.dirty = False
```
Right [after setting the source](https://github.com/jupyterlab/jupyterlab/blob/c4c6aa4146eda3d77c22d29bbb3492100471f7c4/jupyterlab/handlers/ydoc_handler.py#L93), but we need https://github.com/jupyter-server/jupyter_ydoc/pull/16 first.